ooniprobe/Model/Settings/Settings.m

```diff
@@ -23,4 +60,16 @@ -(NSDictionary*)getSettingsDictionary{
     return [mapper dictionaryFromObject:self];
 }
 
+-(NSString*)getSettingsJson{
+    NSError * err;
+    NSData * jsonData = [NSJSONSerialization dataWithJSONObject:[self getSettingsDictionary] options:0 error:&err];
+    if (!err)
+        return [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
+    return nil;
+}
+
+-(id<ExperimentSettings>)toExperimentSettings {
+    return [[ExperimentSettingsAdapter alloc] initWithJson:[self getSettingsJson] settings:self];
```

Comment:
I do not understand here why you get the settings dictionary, then serialise it to JSON, and then immediately parse it again to get a settings adapter. What is the purpose of serialising and parsing?

Reply:
Because I used the same structure `ExperimentSettings` used in Android that is initialized with a `Settings` object and a Json string.

Reply:
If your question was shy I do Settings->NSDictionary->JsonString to initialize the `ExperimentSettings` is because I haven't found a reliable way to do Settings->JsonString